module/documents/actor.js

```diff
@@ -91,79 +92,51 @@ export default class Actor5e extends Actor {
-        if (transformOptions.mergeSaves) originalSaves = original.system.abilities;
-        if (transformOptions.mergeSkills) originalSkills = original.system.skills;
+      if ( original ) {
+        if ( transformOptions.mergeSaves ) originalSaves = original.system.abilities;
+        if ( transformOptions.mergeSkills ) originalSkills = original.system.skills;
       }
     }
 
-    // Ability modifiers and saves
+    // Prepare abilities, skills, & spellcasting
     const globalBonuses = this.system.bonuses || {};
     const bonusData = this.getRollData();
-    const joat = flags.jackOfAllTrades;
-    const dcBonus = this._simplifyBonus(globalBonuses.spell?.dc, bonusData);
-    const saveBonus = this._simplifyBonus(globalBonuses.abilities?.save, bonusData);
-    const checkBonus = this._simplifyBonus(globalBonuses.abilities?.check, bonusData);
-    for (let [id, abl] of Object.entries(this.system.abilities)) {
-      if ( flags.diamondSoul ) abl.proficient = 1;  // Diamond Soul is proficient in all saves
-      abl.mod = Math.floor((abl.value - 10) / 2);
-
-      const isRA = this._isRemarkableAthlete(id);
-      abl.checkProf = new Proficiency(this.system.attributes.prof, (isRA || joat) ? 0.5 : 0, !isRA);
-      const saveBonusAbl = this._simplifyBonus(abl.bonuses?.save, bonusData);
-      abl.saveBonus = saveBonusAbl + saveBonus;
-
-      abl.saveProf = new Proficiency(this.system.attributes.prof, abl.proficient);
-      const checkBonusAbl = this._simplifyBonus(abl.bonuses?.check, bonusData);
-      abl.checkBonus = checkBonusAbl + checkBonus;
-
-      abl.save = abl.mod + abl.saveBonus;
-      if ( Number.isNumeric(abl.saveProf.term) ) abl.save += abl.saveProf.flat;
-      abl.dc = 8 + abl.mod + this.system.attributes.prof + dcBonus;
-
-      // If we merged saves when transforming, take the highest bonus here.
-      if (originalSaves && abl.proficient) {
-        abl.save = Math.max(abl.save, originalSaves[id].save);
-      }
-    }
+    const checkBonus = simplifyBonus(globalBonuses.abilities?.check, bonusData);
+    this._prepareAbilities(bonusData, globalBonuses, checkBonus, originalSaves);
+    this._prepareSkills(bonusData, globalBonuses, checkBonus, originalSkills);
+    this._prepareSpellcasting();
 
     // Attuned items
     if ( this.type !== "vehicle" ) this.system.attributes.attunement.value = this.items.filter(i => {
       return i.system.attunement === CONFIG.DND5E.attunementTypes.ATTUNED;
     }).length;
 
-    // Inventory encumbrance
-    this.system.attributes.encumbrance = this._computeEncumbrance();
-
-    // Prepare skills
-    this._prepareSkills(bonusData, checkBonus, originalSkills);
-
-    // Determine Initiative Modifier
-    this._computeInitiativeModifier(checkBonus, bonusData);
-
-    // Reset class scaling and compute scaling values
-    this._classes = undefined;
-    this._computeScaleValues();
+    // Prepare armor class data
+    this.system.attributes.ac = this.constructor.computeArmorClass(
```

Comment:
I think this method should just be `_prepareArmorClass()` similar to other actor preparation helpers, enabled by the workflow change to clone the original actor as part of AC configuration.